src/tests/controllers/testsController.js

```diff
@@ -49,8 +50,15 @@ async function getTest(req, res, next) {
 
 async function deleteTest(req, res, next) {
     try {
-        await manager.deleteTest(req.params.test_id, req.body);
-        return res.status(200).json();
+        const testsJobs = await jobManager.getJobBasedOnTestId(req.params.test_id);
+        if (testsJobs.length === 0) {
+            await manager.deleteTest(req.params.test_id, req.body);
+            return res.status(200).json();
+        } else {
+            let error = 'Please delete all scheduled jobs for the test before deleting the test';
+            return res.status(500).json({message:error});
+        }
```

Comment:
missing tests (unit tests and integration)

Reply:
Hi, I am pretty new with building unit tests and integration tests, is there any guidelines/ examples I can follow?

Reply:
hi,
yes there many unit tests and integration tests in this project